File: examples/csharp/nats-stream-client/Format.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace CdcExamples.NatsStream;

/// <summary>
/// StreamMessage trägt dasselbe Nachrichtenschema wie der Publisher
/// (<c>internal/adapters/driven/natsstream</c>, <c>SPEC-024</c>) — hier
/// eigenständig geführt: dieses Beispiel importiert keinen privaten
/// Paketbaum dieses Repositories (<c>SPEC-023</c>).
/// </summary>
public sealed record StreamMessage(
    [property: JsonPropertyName("change_id")] string ChangeId,
    [property: JsonPropertyName("transaction_id")] string TransactionId,
    [property: JsonPropertyName("source_table_id")] string SourceTableId,
    [property: JsonPropertyName("sequence")] long Sequence,
    [property: JsonPropertyName("operation")] string Operation,
    [property: JsonPropertyName("old_image")] JsonElement OldImage,
    [property: JsonPropertyName("new_image")] JsonElement NewImage,
    [property: JsonPropertyName("schema_version")] string SchemaVersion,
    [property: JsonPropertyName("schema")] string Schema,
    [property: JsonPropertyName("table")] string Table);

/// <summary>
/// Format baut die Ausgabezeile einer empfangenen Stream-Nachricht
/// (<c>LH-FA-SST-008</c>): Tabelle, Operation und das neue Row Image, dazu
/// die Kennung, an der sich die Nachricht gegen den Lesezugriffsweg
/// <c>cdc.changes</c> halten lässt (<c>change_id</c>) — dieselbe Form wie
/// beim gRPC-Beispiel (<c>examples/csharp/grpc-client/Format.cs</c>).
/// </summary>
public static class Format
{
    public static string FormatChange(StreamMessage change) =>
        $"nats-stream-client: change_id={change.ChangeId} table={change.Schema}.{change.Table} " +
        $"operation={change.Operation} new_image={change.NewImage.GetRawText()}";
}

File: examples/nats-stream-client/format.go
package main

import (
	"encoding/json"
	"fmt"
)

// streamMessage trägt dasselbe Nachrichtenschema wie der Publisher
// (internal/adapters/driven/natsstream) und der SSE-Adapter (SPEC-021/
// SPEC-024) — hier eigenständig geführt: dieses Beispiel importiert keinen
// privaten Paketbaum dieses Repositories (SPEC-023).
type streamMessage struct {
	ChangeID      string          `json:"change_id"`
	TransactionID string          `json:"transaction_id"`
	SourceTableID string          `json:"source_table_id"`
	Sequence      int64           `json:"sequence"`
	Operation     string          `json:"operation"`
	OldImage      json.RawMessage `json:"old_image"`
	NewImage      json.RawMessage `json:"new_image"`
	SchemaVersion string          `json:"schema_version"`
	Schema        string          `json:"schema"`
	Table         string          `json:"table"`
}

// formatChange baut die Ausgabezeile einer empfangenen Stream-Nachricht
// (LH-FA-SST-008): Tabelle, Operation und das neue Row Image, dazu die
// Kennung, an der sich die Nachricht gegen den Lesezugriffsweg cdc.changes
// halten lässt (change_id) — dieselbe Form wie beim gRPC-Beispiel
// (examples/grpc-client/format.go).
func formatChange(c streamMessage) string {
	return fmt.Sprintf("nats-stream-client: change_id=%s table=%s.%s operation=%s new_image=%s",
		c.ChangeID, c.Schema, c.Table, c.Operation, c.NewImage)
}
